File: leetcode_python/leetcode104.py
class tree:
    def __init__(self, val=0, right=None, left=None):
        self.val = val
        self.right = right
        self.left = left


def recusion(root):
    if root is None:
        return 0
    left = recusion(root.left)
    right = recusion(root.right)
    return 1 + max(left, right)


a = tree(5)
a.left = tree(2)
a.right = tree(4)
a.left.left = tree(1)
a.right.left = tree(3)
a.right.right = tree(-1)
a.left.left.left = tree(5)
a.left.left.right = tree(1)
a.right.left.right = tree(6)
a.right.right.right = tree(8)
print(recusion(a))

File: leetcode_c++/leetcode104.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
};
int maxDepth(TreeNode *root)
{
    if (root == nullptr)
        return 0;
    return max(maxDepth(root->left), maxDepth(root->right)) + 1;
}
int main()
{
    TreeNode *root = new TreeNode(3);

    root->left = new TreeNode(9);
    root->right = new TreeNode(20);

    root->right->left = new TreeNode(15);
    root->right->right = new TreeNode(7);
    cout << maxDepth(root) << endl;
}